src/components/cluster/__tests__/cluster_managament.js

```diff
@@ -0,0 +1,176 @@
+import 'jest-dom/extend-expect';
+import { fireEvent, wait } from '@testing-library/react';
+import {
+  clusterCreate as mockClusterCreate,
+  clusterDelete as mockClusterDelete,
+  clusterPatch as mockClusterPatch,
+} from 'actions/clusterActions';
+import { renderRouteWithStore } from 'test_utils/renderRouteWithStore';
+import initialState from 'test_utils/initialState';
+import statusState from 'test_utils/statusState';
+
+// Mock actions to return nothing, we don't want them perform API calls for now,
+// and we don't want them to return any values either.
+// TODO To be changed when mocked http calls ready.
+jest.mock('actions/userActions');
+jest.mock('actions/organizationActions');
+jest.mock('actions/clusterActions', () => {
+  return {
+    clustersLoad: jest.fn(() => () => Promise.resolve()),
+    clusterLoadDetails: jest.fn(() => () => Promise.resolve()),
+    clusterLoadKeyPairs: jest.fn(() => () => Promise.resolve()),
+    clusterCreate: jest.fn(() => () => Promise.resolve()),
+    clusterDelete: jest.fn(() => () => Promise.resolve()),
+    clusterPatch: jest.fn(() => () => Promise.resolve()),
+    clusterLoadDetailsSuccess: jest.fn(() => () => Promise.resolve()),
+  };
+});
+jest.mock('actions/releaseActions');
+jest.mock('actions/catalogActions');
+
+it('drives us to the cluster creation form when launch new cluster button is clicked', async () => {
+  const div = document.createElement('div');
+  const { getByText, getByTestId } = renderRouteWithStore('/', div);
+
+  await wait(() => {
+    const button = getByText(/launch new cluster/i);
+    fireEvent.click(button);
+  });
+
+  expect(getByTestId('cluster-creation-view')).toBeInTheDocument();
+});
+
+it(`renders the form when in new cluster route with default values and calls
```

Comment:
this is kinda testing two things, and we're checking on things which aren't really under the setup of this test.

Like:  `expect(inputs[0]).toHaveValue('My cluster');`

It's ok to check if the form loads well and shows up right, in my tests I also check for output that isn't really under test, as a way to ensure I am on the page I expect. 

But I try not to lock down too much detail.

If there was a way for us to set the default new cluster name as part of the config, and you can manipulate that config at the start of this test ,and then verify that the form comes out as expected, then it would make sense to check on all of those things.. but in this case, I think we're doing a few too many expectations on things that could change, and wouldn't really mean this test should fail for that.

Reply:
Yeah I agree that this test has to be simplified and maybe splitted into two. Rethinking it now.

I don't think we are going to have problems identifying what's going wrong in a test when it fails. Even if it is long.

Reply:
I've  changed it a bit, anyway I will iterate on them because I have to look for rendered elements instead of checking the action creator is called now that we have `nock` and what you've done with it in the mix. Not merging for now.